common/setup.c

```diff
@@ -101,7 +101,13 @@ void __text_init cmdline_parse(const char *cmdline) {
 
             switch (param->type) {
             case STRING:
-                strcpy(param->var, optval);
+                strncpy(param->var, optval, param->varlen);
+                if (strlen(optval) >= param->varlen) {
+                    ((char *) param->var)[param->varlen - 1] = '\0';
+                    printk("WARNING: The commandline parameter value for %s does not fit "
+                           "into the preallocated buffer (size %d >= %d)\n",
```

Comment:
`strlen` return `size_t` so `%lu` is needed and `varlen` is `unsigned int` so `%u` is needed